### DISCUSSION TITLE
Can't use Seleniumbase UC mode with edge using proxies

### DISCUSSION THREAD

**Discussion by aset9803** [seleniumbase/SeleniumBase#2671] [Q&A] [answered]
My code:
driver = Driver(uc=True,browser='edge',headless=False,agent=u_agent,proxy="proxy")
driver.get("https://pixelscan.net")
Please help
![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/592e655a-9e38-468d-9d1c-6849342584fe)

**Comment by mdmintz** [comment, marked as answer]
UC Mode is not supported on Edge.
Only Chrome, Brave, and Opera: https://github.com/seleniumbase/SeleniumBase/discussions/2424#discussioncomment-8100989

**Reply by aset9803** [reply]
I have one more problem. I was able to change the webgl parameters with selenium_stealth till 2 days ago. Suddenly this has stopped working for me. I am usign selenium_stealth and fake user agent
  elif httpagentparser.simple_detect(u_agent)[1].__contains__('Chrome'):
                                driver = Driver(uc=True,headless=False,undetectable=True,proxy="proxy",agent=u_agent)
                                stealth(
                                driver,                                                                                               
                                languages= ["en-US", "en"],
                                vendor = "Microsoft Inc.",
                                platform = 'Win',
                                webgl_vendor = "Intel Inc.",
                                renderer = "Intel Iris OpenGL Engine"
                                                                )
  print(httpagentparser.simple_detect(u_agent))
                driver.uc_open('https://pixelscan.net') 

![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/d0d650c6-8b02-4603-8e0a-b8ff9ff37e6a)

**Reply by aset9803** [reply]
Update: The values work with driver.default_get(). But is it safe to use since I am hiding my bot?

**Reply by mdmintz** [reply]
Depends. Here's the script I use for Pixelscan:

[SeleniumBase/examples/raw_pixelscan.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_pixelscan.py)

```python
from seleniumbase import SB

with SB(uc=True, incognito=True, test=True) as sb:
    sb.driver.uc_open_with_reconnect("https://pixelscan.net/", 10)
    sb.remove_elements("jdiv")  # Remove chat widgets
    sb.assert_text("No automation framework detected", "pxlscn-bot-detection")
    sb.assert_text("You are not masking your fingerprint")
    sb.highlight("span.text-success", loops=8)
    sb.sleep(1)
    sb.highlight("pxlscn-fingerprint-masking div", loops=9, scroll=False)
    sb.sleep(1)
    sb.highlight("div.bot-detection-context", loops=10, scroll=False)
    sb.sleep(2)
```

**Reply by aset9803** [reply]
thanks for the info. I have some questions since I have experimented a lot with seleniumbase now.
1. Can I enable touch events in mobile mode where a circle appears for touch? Like this in undetectable browser:- 
![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/91100073-ee2c-4575-8ac5-f5c5b7d5e9ea)
2. When I am passing user agent, it shows red in pixelscan but it changes the user agent. It shows me that you are using firefox on ios but actually i am using chrome on windows. But it turns red. But in undetectable, it changes to green. :- 
![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/dbfcd8c7-5e6f-41fa-bd54-4feb432937ed)
my driver - 
![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/d0b92ef6-3557-4130-a53c-6937b768b482)

3. is there anyway to spoof the hardware concurrency in pixelscan like undetectable does.:- 
![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/5444e771-6d46-44ac-8989-6c7f4d91c10e)

4. is there anyway to change the webgl version. I am able to change platform, vendor and renderer through stealth. :- 
![image](https://github.com/seleniumbase/SeleniumBase/assets/150322527/3e7bd1cd-44a7-43a4-a125-0abc08189192)
i want to change this to webgl 1.0 only(for iphones)

thanks in advance.

**Reply by mdmintz** [reply]
SeleniumBase Mobile Mode is just a simple mobile device emulator that doesn't have all the functionality that Appium has, which means that touch actions and other such things aren't supported (unless you find something on Stack Overflow).

For the Pixelscan test, this is the baseline script:

```python
from seleniumbase import SB

with SB(uc=True, incognito=True, test=True) as sb:
    sb.driver.uc_open_with_reconnect("https://pixelscan.net/", 10)
    sb.remove_elements("jdiv")  # Remove chat widgets
    sb.assert_text("No automation framework detected", "pxlscn-bot-detection")
    not_masking = "You are not masking your fingerprint"
    sb.assert_text(not_masking, "pxlscn-fingerprint-masking")
    sb.highlight("span.text-success", loops=8)
    sb.sleep(1)
    sb.highlight("pxlscn-fingerprint-masking div", loops=9, scroll=False)
    sb.sleep(1)
    sb.highlight("div.bot-detection-context", loops=10, scroll=False)
    sb.sleep(2)
```

I continually check that script to make sure things still work. If you make modifications, (such as changing the user_agent, etc), then the script is no longer guaranteed to work.

WebGL has already been discussed here: https://github.com/seleniumbase/SeleniumBase/discussions/2248#discussioncomment-7501491

For the `selenium-stealth` integration, see: https://github.com/seleniumbase/SeleniumBase/issues/2404

**Reply by aset9803** [reply]
Hi. Thanks. I have already seen these two threads. My question was whether there is a way to change the hardware concurrency and web gl version shown in pixelscan.

**Reply by mdmintz** [reply]
If the `selenium-stealth` integration doesn't do it (https://github.com/seleniumbase/SeleniumBase/issues/2404), then check [Stack Overflow](https://stackoverflow.com/).
